packages/jellyfish-api-core/__tests__/category/masternode/getMasternodeBlocks.test.ts

```diff
@@ -0,0 +1,117 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { MasternodeBlock } from '../../../src/category/masternode'
+import { Testing } from '@defichain/jellyfish-testing'
+
+describe('Masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should show list of blocks for masternode', async () => {
+    const address = await testing.container.getNewAddress('', 'legacy')
+    const id = await testing.container.call('createmasternode', [address])
+    await testing.container.generate(1)
+
+    await testing.container.setDeFiConf(['masternode_operator=' + address])
+    await testing.container.restart()
+
+    await testing.container.generate(20)
+    const mintedBlockNumber = 5
+    await testing.container.generate(mintedBlockNumber, address)
+
+    const identifier: MasternodeBlock = {
+      id
+    }
+    let masternodeBlock = await testing.rpc.masternode.getMasternodeBlocks(identifier)
+    expect(Object.keys(masternodeBlock).length).toEqual(mintedBlockNumber)
+
+    identifier.ownerAddress = address
+    identifier.id = undefined
+    masternodeBlock = await testing.rpc.masternode.getMasternodeBlocks(identifier)
+    expect(Object.keys(masternodeBlock).length).toEqual(mintedBlockNumber)
+
+    identifier.operatorAddress = address
+    identifier.ownerAddress = undefined
+    masternodeBlock = await testing.rpc.masternode.getMasternodeBlocks(identifier)
+    expect(Object.keys(masternodeBlock).length).toEqual(mintedBlockNumber)
+  })
+
+  it.skip('should show only list of blocks for masternode according to the depth', async () => {
```

Comment:
@chee-chyuan you can remove the skip here and check this against https://github.com/DeFiCh/ain/pull/946
Use the image from that branch. find it here -> https://github.com/DeFiCh/ain/runs/4442656931?check_suite_focus=true#step:7:81
If all is good, please make the PR ready to review and ping in the channel.